packages/eslint-plugin/tests/rules/space-infix-ops.test.ts

```diff
@@ -278,332 +384,763 @@ ruleTester.run('space-infix-ops', rule, {
       output: `
-        class Test {
-          value: string | number;
-        }
+        type Test =
+        | string
+        | (() => void);
       `,
       errors: [
         {
           messageId: 'missingSpace',
-          column: 25,
+          column: 9,
           line: 3,
         },
       ],
     },
     {
       code: `
-        class Test {
-          value: string& number;
-        }
+        type Test =
+        &string
+        & number;
       `,
       output: `
-        class Test {
-          value: string & number;
-        }
+        type Test =
+        & string
+        & number;
       `,
       errors: [
         {
           messageId: 'missingSpace',
-          column: 24,
+          column: 9,
           line: 3,
         },
       ],
     },
     {
       code: `
-        class Test {
-          value: string| number;
-        }
+        type Test =
+        &string
+        & (() => void);
       `,
       output: `
-        class Test {
-          value: string | number;
-        }
+        type Test =
+        & string
+        & (() => void);
+      `,
+      errors: [
+        {
+          messageId: 'missingSpace',
+          column: 9,
+          line: 3,
+        },
+      ],
+    },
+    {
+      code: `
+        interface Test {
+          prop: string| number;
+        }
+      `,
+      output: `
+        interface Test {
+          prop: string | number;
+        }
+      `,
+      errors: [
+        {
+          messageId: 'missingSpace',
+          column: 23,
+          line: 3,
+        },
+      ],
+    },
+    {
+      code: `
+        interface Test {
+          prop: string| (() => void);
+        }
+      `,
+      output: `
+        interface Test {
+          prop: string | (() => void);
+        }
+      `,
+      errors: [
+        {
+          messageId: 'missingSpace',
+          column: 23,
+          line: 3,
+        },
+      ],
+    },
+    {
+      code: `
+        interface Test {
+          prop: string |number;
+        }
+      `,
+      output: `
+        interface Test {
+          prop: string | number;
+        }
+      `,
+      errors: [
+        {
+          messageId: 'missingSpace',
+          column: 24,
+          line: 3,
+        },
+      ],
+    },
+    {
+      code: `
+        interface Test {
+          prop: string |(() => void);
+        }
+      `,
+      output: `
+        interface Test {
+          prop: string | (() => void);
+        }
+      `,
+      errors: [
+        {
+          messageId: 'missingSpace',
+          column: 24,
+          line: 3,
+        },
+      ],
+    },
+    {
+      code: `
+        interface Test {
+          prop: (() => void) |(() => void);
+        }
+      `,
+      output: `
+        interface Test {
+          prop: (() => void) | (() => void);
+        }
+      `,
+      errors: [
+        {
+          messageId: 'missingSpace',
+          column: 30,
+          line: 3,
+        },
+      ],
+    },
```

Comment:
Could you add a few test cases for functions that are wrapped multiple times? `((() => void))` and the like? Parenthesis and wrapping expressions are tricky.

Reply:
That are nice edge cases. The current fix didn't consider this scenario with function type having multiple parenthesis wrapping. I think I need to revisit the fix again with this scenario in mind.

Reply:
@JoshuaKGoldberg Handled edge cases you have pointed out in the latest commit. Added more test cases to check the same edge cases.